### PR TITLE
[ADD] module logistic_order_multicurrency

### PR DESCRIPTION
**Pull request by yvaucher**
This module shows amounts in company currency bellow total amount on logistic order
It also show currency in list view.

### CODE REVIEW

File: logistic_order_multicurrency/model/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    _inherit = 'sale.order'
+
+    @api.one
+    @api.depends('amount_total',
+                 'currency_id',
+                 'company_id.currency_id')
+    def _compute_prices_in_company_currency(self):
+        """ Compute amount_total in company currency
+
+        Date for conversion is today
```
Comment (gurneyalex) [inline logistic_order_multicurrency/model/sale_order.py]: are we sure we want to use today, always? I would be tempted have this depend on the state of the SO: maybe use today in draft mode, and the day of confirmation of the SO after. Or maybe we want today until the SO is paid?

@jgrandguillaume: opinion?

**Comment by gurneyalex** [comment]
could you add an automated test?

File: logistic_order_multicurrency/model/sale_order.py (hunk 1 above)
Comment (jgrandguillaume) [inline logistic_order_multicurrency/model/sale_order.py]: I said at "Date" where date is the order date.. Sorry if it wasn't clear enough

**Comment by yvaucher** [comment]
@gurneyalex added tests

**Comment by gurneyalex** [comment]
error in test code, I think
vertical-ngo/logistic_order_multicurrency/tests/test_amount_total_co.py", line 61, in test_amount_other_currency

--> AttributeError: 'module' object has no attribute 'now'

**Comment by yvaucher** [comment]
@gurneyalex fixed

File: logistic_order_multicurrency/i18n/logistic_order_multicurrency.pot [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: logistic_order_multicurrency
+#: model:ir.model,name:logistic_order_multicurrency.model_sale_order
+msgid "Sales Order"
+msgstr ""
+
```
Comment (gurneyalex) [inline logistic_order_multicurrency/i18n/logistic_order_multicurrency.pot:33]: This file looks out of date (there is no mention of the added fields in the module).

**Comment by jgrandguillaume** [comment]
Tested. +1 when .pot file is updated

**Comment by lepistone** [comment]
:+1: for me, too. @yvaucher ping me for a merge when the translation is up to date. Thanks!

**Comment by yvaucher** [comment]
updated pot file

**Comment by guewen** [comment]
@yvaucher seems that some values still miss in the pot file, I don't see

```
+ string="Total in Company currency",
+ help="Total converted to company currency at today's rate."
```

**Comment by yvaucher** [comment]
@guewen Odoo didn't generated them :disappointed: 

Maybe it is because they aren't displayed in view.

File: logistic_order_multicurrency/model/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    amount_total_co = fields.Float(
+        compute='_compute_prices_in_company_currency',
+        string="Total in Company currency",
+        help="Total converted to company currency at today's rate."
+        )
+
+    company_currency_id = fields.Many2one(
+        related='company_id.currency_id',
+        comodel_name='res.currency',
+        string='Currency')
```
Comment (ehdem) [inline logistic_order_multicurrency/model/sale_order.py:52]: Add help option

File: logistic_order_multicurrency/model/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    # define a dummy stored field for groupby
+    # to be able to use new auto trigger store
+    # and to avoid AssertionError: Fields in 'groupby' must be regular
+    # database-persisted fields (no function or related fields), or function
+    # fields with store=True
+    stored_currency_id = fields.Many2one(
+        related='pricelist_id.currency_id',
+        comodel_name='res.currency',
+        string='Currency',
+        store=True)
```
Comment (ehdem) [inline logistic_order_multicurrency/model/sale_order.py:68]: Add help option

Comment (yvaucher) [inline logistic_order_multicurrency/model/sale_order.py:68]: This field is not even displayed so an help text is useless

File: logistic_order_multicurrency/model/sale_order.py (hunk 3 above)
Comment (yvaucher) [inline logistic_order_multicurrency/model/sale_order.py:52]: I don't think it is necessary `company_currency_id` is hidden in view and only used in money widget.

**Comment by yvaucher** [comment]
@ehdem Hello thanks for the review, it is a good idea to make sure help texts are added to fields. However, in current case, those fields are there for computation and using money widget. They aren't displayed to the user. So a user wouldn't have access to that help message.

**Comment by lepistone** [comment]
I tried myself to generate the translation files on a clean database, and I get the exact same result (I took the occasion to add a french stub). So @guewen I don't see any problem here, and don't know why a few strings seem missing.

I'm merging as soon as the build is green. Thanks to all!

**Comment by lepistone** [comment]
Like we discussed on #115, I updated `fr.po` to have empty strings.

**Comment by yvaucher** [comment]
@lepistone thanks!

**Comment by ehdem** [comment]
@yvaucher: thank, you are right, but I think it is a way to help developper to know the field definition.

**Pull request merged**